### PR TITLE
chore(lint): Switch from JSHint to ESLint

### PR DESCRIPTION
**Pull request by pdehaan**

### CODE REVIEW

**Comment by TDA** [comment]
LGTM! but travis seems to have blown up. You have made travis angry @pdehaan :D :D

File: package.json [hunk 1]
```diff
@@ -1,36 +1,44 @@
...
   "author": "Mozilla (https://mozilla.org/)",
-  "homepage": "https://github.com/mozilla/fxa-auth-db-mem",
   "bugs": {
     "url": "https://github.com/mozilla/fxa-auth-db-mem/issues"
   },
-  "license": "MPL-2.0",
   "dependencies": {
     "bluebird": "2.2.2",
-    "convict": "0.6.1",
+    "convict": "0.8.0",
```
Comment (vladikoff) [inline package.json]: @pdehaan updating this version is too dangerous here, this is not related to this PR either.  Should probably not be done here

File: config/config.js [hunk 2]
```diff
@@ -46,7 +46,7 @@ module.exports = function (fs, path, url, convict) {
 
   var files = envConfig.split(',').filter(fs.existsSync)
   conf.loadFile(files)
-  conf.validate()
```
Comment (vladikoff) [inline config/config.js]: same as the other comment, convict changes are not related to the PR
